packages/react/src/hooks/transactions/useWatchPendingTransactions.ts

```diff
@@ -0,0 +1,36 @@
+import type { Transaction } from 'ethers'
+import * as React from 'react'
+
+import { useProvider, useWebSocketProvider } from '../providers'
+import { useChainId } from '../utils'
+
+export type UseWatchPendingTransactionsConfig = {
+  /** The chain ID to listen on. */
+  chainId?: number
+  /** Subscribe to changes */
+  enabled?: boolean
+  /** Function fires when a pending transaction enters the mempool. */
+  listener: (transaction: Transaction) => void
+}
+
+export function useWatchPendingTransactions({
+  chainId: chainId_,
+  enabled = true,
```

Comment:
this option seems redundant. You either subscribe to to events by using the hook or you don't.

Reply:
There may be a case where a consumer wants to disable subscriptions based on a depdendant state variable. wdyt?

Reply:
Sure, yeah. That makes sense